### PR TITLE
add resharding unit test 

### PR DESCRIPTION
**Pull request by chaosma**
added resharding unit tests

### CODE REVIEW

File: core/resharding_test.go [hunk 1]
```diff
@@ -3,16 +3,47 @@ package core
...
-		fmt.Printf("ShardID: %v, NodeList: %v\n", ss[i].ShardID, ss[i].NodeList)
-	}
-}
+	"github.com/harmony-one/harmony/core/types"
+	"github.com/stretchr/testify/assert"
+)
 
 func TestFakeNewNodeList(t *testing.T) {
 	nodeList := fakeNewNodeList(42)
 	fmt.Println("newNodeList: ", nodeList)
```
Comment (mikedoan) [inline core/resharding_test.go]: why is it a test?

File: core/resharding_test.go [hunk 2]
```diff
@@ -3,16 +3,47 @@ package core
...
+	cpList = append(cpList, nodeList...)
+	Shuffle(nodeList)
+	cnt := 0
+	for i := 0; i < 10; i++ {
+		if cpList[i] == nodeList[i] {
+			cnt++
+		}
+	}
+	if cnt == 10 {
+		t.Error("Shuffle list is the same as original list")
```
Comment (mikedoan) [inline core/resharding_test.go]: can you clarify what you are trying to test here

File: core/resharding_test.go (hunk 1 above)
Comment (chaosma) [reply core/resharding_test.go]: Since it's random, I just by looking at the result, can be removed

File: core/resharding_test.go (hunk 2 above)
Comment (chaosma) [reply core/resharding_test.go]: When shuffle a list, we have a very low probability that the list stay the same

Comment (mikedoan) [reply core/resharding_test.go]: maybe should do another type of checking

**Pull request merged**